tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -500,3 +509,24 @@ def parse_iso8601_retention_period(iso8601_retention_period):
         units = next(units for units in RetentionPeriod.Units
             if units.iso8601_symbol == iso8601_symbol)
         return period, units
+
+
+@lru_cache()
```

Comment:
Is it needed?  Just one query per session init?  Breaks container-host, I'll have to clear it periodically to ensure secret updates are reflected in long running processes.